izer/yamlcfg.py

```diff
@@ -74,15 +74,31 @@ def error_exit(message, sequence):
 
     print(f'Reading {config_file} to configure network...')
 
-    # Run yamllint first
     if not os.path.exists(config_file):
         eprint(f'YAML configuration file {config_file} does not exist!')
 
-    yaml_config = yamllint.config.YamlLintConfig('extends: relaxed')
-    with open(config_file, mode='r', encoding='utf-8') as cfg_file:
-        for p in yamllint.linter.run(cfg_file, yaml_config):
-            eprint(f'{config_file} line {p.line}, col {p.column}: {p.desc}',
-                   error=p.level == 'error')
+    # Run linter first if configured
+    if linter:
+        try:
+            proc = subprocess.run(
+                [linter, config_file],
+                shell=False,
+                text=True,
+                capture_output=True,
+                timeout=5.0,
+                check=False,
+            )
+            errors = proc.stderr.splitlines()
+            for i, line in enumerate(errors):
+                eprint(line, exit_code=None if i < len(errors) - 1 else 1)
+            warnings = proc.stdout.splitlines()
+            for w in warnings:
+                if ' error ' in w:
+                    eprint('  ' + w)
+                elif w != '':
+                    wprint(w)
+        except FileNotFoundError:
+            wprint(f'Cannot run "{linter}" linter to check {config_file}')
```

Comment:
Should we add an else condition to warn the user that the config check is skipped because the linter is not configured?  Or don't we need such a warning since this condition is seen only when args.yamllint is set to None deliberately?

Reply:
It actually cannot even be set to None (since the argument is a string). It's just for future-proofing, so the warning will always happen.